packages/databases-collections/src/stores/collections-store.js

```diff
@@ -95,7 +99,23 @@ store.onActivated = (appRegistry) => {
    *
    * @param {String} ns - The namespace.
    */
-  appRegistry.on('select-database', onSelectDatabase);
+  appRegistry.on('select-database', (dbName) => {
+    onSelectDatabase(dbName);
+  });
+
+  /**
+   * In all other cases reset current database name so that plugin is aware that
+   * it is not active
+   */
+  appRegistry.on('open-instance-workspace', () => {
+    onSelectDatabase();
+  });
+  appRegistry.on('open-namespace-in-new-tab', () => {
+    onSelectDatabase();
+  });
+  appRegistry.on('select-namespace', () => {
+    onSelectDatabase();
+  });
```

Comment:
Not resetting this was causing a fun bug in Compass where opening a database first and then opening a bunch of collections (even from different namespaces) will close all your tabs when any collection from the this last opened database was dropped